src/utils/eth_transaction.cairo

```diff
@@ -57,19 +57,25 @@ namespace EthTransaction {
         // Verify signature
         // The signature is at the end of the rlp encoded list and takes
         // 5 bytes for v = 1 byte for len (=4) + 4 bytes for {0,1} + CHAIN_ID * 2 + 35
-        // 33 bytes for r = 1 byte for len (=32) + 32 bytes for r word
-        // 33 bytes for s = 1 byte for len (=32) + 32 bytes for s word
+        // 1 + r_len bytes for r = 1 byte for len (<= 32) + length in bytes for r word
+        // 1 + s_len bytes for s = 1 byte for len (<= 32) + length in bytes for s word
         // This signature_len depends on CHAIN_ID, which is currently 0x 4b 4b 52 54
-        local signature_len = 1 + 4 + 1 + 32 + 1 + 32;
         local signature_start_index = 6;
+        tempvar r_len = sub_items[signature_start_index + 1].data_len;
+        tempvar s_len = sub_items[signature_start_index + 2].data_len;
```

Comment:
do you need to `tempvar`, isn't `let` enough?

Reply:
will try `let`, is it correct to say that `tempvar` is more expensive in terms of steps because it stores the value in a new cell of `ap`, where as `let` is resolved at compilation?